File: ios/nplus/GoogleAds/RNGAMBannerViewManager.m
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(RNGAMBannerViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(adUnitId, NSString)
RCT_EXPORT_VIEW_PROPERTY(adSize, NSString)
RCT_EXPORT_VIEW_PROPERTY(onAdLoaded, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onAdFailedToLoad, RCTDirectEventBlock)

RCT_EXTERN_METHOD(loadAd:(nonnull NSNumber *)reactTag)

@end

File: ios/nplus/GoogleAds/RNGAMBannerViewManager.swift
import Foundation
import React

@objc(RNGAMBannerViewManager)
class RNGAMBannerViewManager: RCTViewManager {
    
    override func view() -> UIView! {
        return RNGAMBannerView()
    }
    
    override static func requiresMainQueueSetup() -> Bool {
        return true
    }
    
    @objc func loadAd(_ reactTag: NSNumber) {
        DispatchQueue.main.async {
            if let view = self.bridge?.uiManager.view(forReactTag: reactTag) as? RNGAMBannerView {
                view.loadAd()
            }
        }
    }
}
